### PR TITLE
[AGENT-4670] Implement extend Airflow Operators to set-up a datetime partitioning project

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Added Airflow sample DAG for datetime partitioning project (OTV)
Most of work was done in scope of [AGENT-4669] Implement/extend Airflow Operators to set up a Time Series project

## Rationale
Users should be able to setup datetime partitioning project using Airflow Operators

[AGENT-4669]: https://datarobot.atlassian.net/browse/AGENT-4669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: datarobot_provider/example_dags/datarobot_datetime_partitioning_pipeline_dag.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+@dag(
+    schedule=None,
+    start_date=datetime(2022, 1, 1),
+    tags=['example', 'datetime_partitioning'],
+    params={
+        "dataset_file_path": "/train_datetime.csv",
+        "project_name": "test airflow datetime-partitioning project",
+        "autopilot_settings": {
+            "target": "y",
```
Comment (elatt) [inline datarobot_provider/example_dags/datarobot_datetime_partitioning_pipeline_dag.py:28]: Do we provide any of these sample datasets or do we expect it to be clear enough as an example?

File: datarobot_provider/example_dags/datarobot_datetime_partitioning_pipeline_dag.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            "mode": AUTOPILOT_MODE.QUICK,
+        },
+        "datetime_partitioning_settings": {
+            "use_time_series": False,
+            "datetime_partition_column": 'datetime',
+            "number_of_backtests": 1,
+            "autopilot_data_selection_method": DATETIME_AUTOPILOT_DATA_SELECTION_METHOD.DURATION,
+            "validation_duration": "P1Y0M0D",
+            "gap_duration": "P0Y0M0D",
```
Comment (elatt) [inline datarobot_provider/example_dags/datarobot_datetime_partitioning_pipeline_dag.py:37]: I'm not familiar with these `P0Y0M0D` values. Probably people new to the DR API would be unfamiliar too... maybe adding a link to our public docs describing these values?

**Review by elatt — APPROVED** [review]
LGTM

**Review by alexromanyuk — APPROVED** [review]

File: datarobot_provider/example_dags/datarobot_datetime_partitioning_pipeline_dag.py (hunk 2 above)
Comment (oleksandr-saienko) [reply datarobot_provider/example_dags/datarobot_datetime_partitioning_pipeline_dag.py]: Added methods from datarobot.helpers.partitioning_methods to construct duration string like: construct_duration_string(years=1),

**Pull request merged**